pkg/executor/detach.go

```diff
@@ -115,3 +134,37 @@ func (e *IndexLookUpExecutor) Detach() (exec.Executor, bool) {
 
 	return newExec, true
 }
+
+// Detach detaches the current executor from the session context.
+func (e *ProjectionExec) Detach() (exec.Executor, bool) {
+	// check whether the `Projection` requires any optional property
+	// Now, no optional property is copied, so if it requires any optional property, it should return false.
+	if !e.evaluatorSuit.RequiredOptionalEvalProps().IsEmpty() {
```

Comment:
I think we can add a TODO here that some optional properties can be copied when detach.